src/ashtha/Shop.java

```diff
@@ -0,0 +1,121 @@
+package ashtha;
+
+public class Shop {
+
+	int initialMaggieStocks;
+	int initialDosaStocks;
+	int initialPouchesStocks;
+	int initialPanipuriStocks;
+	int initialMasalaStocks;
+	
+	/*method to set the inventory values*/
+	void setInventory (int maggieStock, int dosaStock, int pouchesStock, int panipuriStock, int masalaStock) {
+		initialMaggieStocks = maggieStock;
+		initialDosaStocks = dosaStock;
+		initialPouchesStocks = pouchesStock;
+		initialPanipuriStocks = panipuriStock;
+		initialMasalaStocks = masalaStock;
+	}
+	
+	/*method to update the inventory value*/
+	void updateInventory (int requestedMaggieQty, int requestedDosaQty, int requestedPouchesQty, int requestedPanipuriQty, int requestedMasalaQty, boolean partial) {
+		/* to update maggie stock*/
+		if (initialMaggieStocks >= requestedMaggieQty) {
+			initialMaggieStocks = initialMaggieStocks - requestedMaggieQty;
+		}else if (initialMaggieStocks == 0) {
+			System.out.println("Maggie is currently out of stock");
+		}else if (true) {
+			System.out.println("You requested for "+requestedMaggieQty +" maggie packets but only "+initialMaggieStocks+" packets available in stock. Do you want to proceed with the cart?");
+			System.out.println("Order placed successfully for "+initialMaggieStocks +" maggie packets.");
+			initialMaggieStocks = 0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct